Feature dsl experiment generation with number of users parameter

jeppe-style opened the pull request.

jeppe-style committed:
Updates versions in POM. Adds test for ExperimentTerminationCriteria. Moves previous tests into ‘simone’ subfolder.

jeppe-style committed:
Adds tests for termination criteria.

jeppe-style committed:
Adds support for goal.exploration_space.workload.users

jeppe-style committed:
Adds method to generate experiment from test and parameters (work-in-progress).

jeppe-style committed:
Adds functionality to generate experiment YAML from test YAML with changing number of users.

VincenzoFerme commented:
@jeppe-style fix https://www.codacy.com/app/BenchFlow/benchflow/pullRequest?prid=612662, or @igormoreno improve the settings if these warnings are ok to have. Thanks!

jeppe-style commented:
Yes, I have changes that will solve these issues but I thought to do it in a second pull-request, basically I will remove the Try and throw an exception instead since that is easier to handle in Java anyway.

VincenzoFerme committed:
Merge branch 'devel' into feature-dsl-experiment-generation

jeppe-style committed:
Updates error handling so that exceptions are thrown outside the library and Try’s are used inside.

jeppe-style commented:
@VincenzoFerme the code is now updated so that the checks pass and now it will be easier to use in Java.

@igormoreno  Some of the issues that are shown in codacy I don't get when running `mvn verify`.

jeppe-style committed:
Removes .codeclimate.yml and .travis.yml.

VincenzoFerme commented on benchflow-dsl/pom.xml:
```diff
@@ -36,13 +38,13 @@
         <dependency>
             <groupId>org.scala-lang.modules</groupId>
-            <artifactId>scala-xml_2.11</artifactId>
+            <artifactId>scala-xml_2.12</artifactId>
```
`2.12` can also be defined as properties, or probably can be even extracted from `<scala.xml.version>` by filtering out the last `.1`. This applies to all the cases in this, and other files.

VincenzoFerme commented on benchflow-dsl/src/test/resources/tests/exploration/users/benchflow-test.yml:
```diff
@@ -0,0 +1,90 @@
+          values: [5, 10, 15, 20]
+
+
+  users: 10
```
Why do you have the setting also here, if you define an exploration space on users?

VincenzoFerme reviewed (changes requested):
This is just a very quick review, we should plan more in depth reviews in next iterations. An issue has been opened to schedule a complete code review (#244)

igormoreno committed:
Small simplification using string interpolation

This change serves as an example of the use of string interpolation.
In some cases it may make your code more clear.

igormoreno reviewed (approved):
- Consider using string interpolation if you need to do 
too many string concatenations. I committed a small example.

- It would be nice to try to get read of those hardcoded strings: "configuration.goal.exploration_space.workload.users.". Finding your way somehow to build them dynamically. But it looks like that's not something trivial. So I would leave it for another iteration

igormoreno committed:
Replace head/tail for take/drop to avoid exception

VincenzoFerme committed:
Merge branch 'devel' into feature-dsl-experiment-generation

VincenzoFerme committed:
Merge branch 'devel' into feature-dsl-experiment-generation

VincenzoFerme committed:
Attemps to fix an issue in Deliver to Bintray

VincenzoFerme committed:
Fake change to trigger the build

VincenzoFerme committed:
Debugging error in delivering to BinTray

VincenzoFerme committed:
Fake change to trigger the build

jeppe-style committed:
Updates pom.xml to make better use of properties. Removes unnecessary key in exploration/users/benchflow-test.yml.

jeppe-style committed:
Merge branch 'feature-dsl-experiment-generation' of https://github.com/benchflow/benchflow into feature-dsl-experiment-generation

* 'feature-dsl-experiment-generation' of https://github.com/benchflow/benchflow:
  Fake change to trigger the build
  Debugging error in delivering to BinTray
  Fake change to trigger the build
  Attemps to fix an issue in Deliver to Bintray
  Moves custom tag generation code to solve an issue
  Migrate Docker Images and (Partially) Upgrade to Wercker Build System (#246)
  Replace head/tail for take/drop to avoid exception
  Small simplification using string interpolation
  Adds files from jeppe-style/performance-test-manager repo (#195)
  Includes files from benchflow/experiments-manager repo (#194)
  Renames a step
  Adds a file to WERCKER_OUTPUT_DIR, to speed up the not needed store step
  Relies on WERCKER_OUTPUT_DIR to speed up store step

jeppe-style committed:
Updates message for unsupported*Operation to make it more readable.

VincenzoFerme committed:
Debugging error in delivering to BinTray

jeppe-style committed:
Merge branch 'feature-dsl-experiment-generation' of https://github.com/benchflow/benchflow into feature-dsl-experiment-generation

* 'feature-dsl-experiment-generation' of https://github.com/benchflow/benchflow:
  Debugging error in delivering to BinTray

VincenzoFerme committed:
Debugging error in delivering to BinTray

igormoreno commented:
Issue regarding hardcoded strings: #249

jeppe-style commented:
Thanks @igormoreno and @VincenzoFerme for your comments! I have now included/fixed them so we should be good to go.

VincenzoFerme committed:
Debugging error in delivering to BinTray

VincenzoFerme committed:
Debugging error in delivering to BinTray

VincenzoFerme committed:
Debugging error in delivering to BinTray

VincenzoFerme committed:
Tests if file exists before attempting curl upload

VincenzoFerme committed:
Publishes only branch version

Commit specific builds are part of the file of the branch version

VincenzoFerme committed:
Attemps to move code back to a dedicated script

VincenzoFerme committed:
Set again execution permission for the script

VincenzoFerme committed:
Copies devops script for subsequent step

VincenzoFerme committed:
Adds a comment about a future possible improvement
